renovate: document troubleshooting option (#1039)
rust-lang-nursery/rust-forge: src/infra/docs/renovate.md

GitHub issue, so that you can
trigger PRs in the repository by interacting with that issue.

### 4. Troubleshooting Renovate behavior

> If you don't understand why Renovate does not behave as you expected,

Reviewer: ```suggestion
If you don't understand why Renovate doesn't behave as you expect,
```

*[View changes since the review](https://triagebot.infra.rust-lang.org/gh-changes-since/rust-lang/rust-forge/1039/cd8499953aa16557e702320c7294ff99770ddafe..9b11bb7052551cc33358b97c37baad6b42f66a92)*

you can run it locally in dry-run mode and evaluate how it resolves
dependency updates:

```bash
docker run --rm -it\

>     -e LOG_LEVEL=="log-level"\              # usually info or debug

Reviewer: ```suggestion
```
I would omit this so that people can copy paste in their terminal directly.
Then document to add `-e LOG_LEVEL="debug"` if they need more logs

*[View changes since the review](https://triagebot.infra.rust-lang.org/gh-changes-since/rust-lang/rust-forge/1039/cd8499953aa16557e702320c7294ff99770ddafe..9b11bb7052551cc33358b97c37baad6b42f66a92)*

    -e RENOVATE_TOKEN="<github-token>"\     # a Github token, e.g. running `gh auth token`

>     -e GITHUB_COM_TOKEN="<github-token>"\   # the same Github token works here

Reviewer: I wonder if we can preface this command with `GITHUB_TOKEN=$(gh auth token) &&\` so that copy pasting it works

*[View changes since the review](https://triagebot.infra.rust-lang.org/gh-changes-since/rust-lang/rust-forge/1039/cd8499953aa16557e702320c7294ff99770ddafe..9b11bb7052551cc33358b97c37baad6b42f66a92)*

Author: Great suggestion!

GitHub issue, so that you can
trigger PRs in the repository by interacting with that issue.

### 4. Troubleshooting Renovate behavior

If you don't understand why Renovate doesn't behave as you expect,
you can run it locally in dry-run mode, and evaluate how it resolves
dependency updates:

```bash

RENOVATE_DRYRUN_TOKEN=$(gh auth token) &&\
docker run --rm -it\
    -e RENOVATE_TOKEN="$RENOVATE_DRYRUN_TOKEN"\
    -e GITHUB_COM_TOKEN="$RENOVATE_DRYRUN_TOKEN"\
    -v /tmp:/tmp\
    -v $PWD:/usr/src/app\

>     renovate/renovate:latest renovate --platform=local --repository-cache=reset --dry-run

Reviewer: ```suggestion
    renovate/renovate:latest renovate --platform=local --repository-cache=reset --dry-run=lookup
```

This change fixes this warning: `cli config dryRun property has been changed to full`.

An alternative is also `--dry-run=full` but it has an additional warning because of `platform=local`

Docs https://docs.renovatebot.com/self-hosted-configuration/#dryrun
